src/routes/chats.js

```diff
@@ -1,25 +1,229 @@
 module.exports = (() => {
 	const express = require('express');
 	const router = express.Router();
-	// const path = require('path');
-	// const chatController = require(path.join(__dirname, '../controllers/chat.controller'));
+	const path = require('path');
+	const chatController = require(path.join(__dirname, '../controllers/chat.controller'));
+	// bring in middleware to verify user's membership in chat
+	const membershipMiddleware = require(path.join(__dirname, '../middlewares/membership'));
 
-	// must have some middleware to verify user is authenticated
-	// AND user making the API call is part of the given chat
+	/**
+	* @api {get} /chat Get chats user is in
+	* @apiName GetChats
+	* @apiGroup Chat
+	*
+	* @apiHeader {String} token Authorization Bearer Token.
+	*
+	* @apiSuccess {Object[]} - List of chats.
+	* @apiSuccess {String} -.id Chats id.
+	*
+	* @apiSuccessExample Success Response:
+	*     HTTP/1.1 200 OK
+	*     [
+	*		{
+	*			"id": 1,
+	*			"name": "Chat 1"
+	*		},
+	*		{
+	*			"id": 4,
+	*			"name": "Chat 4"
+	*		}
+	*	 ]
+	*
+	* @apiError UnauthorizedError Invalid/missing token in authorization header.
+	*
+	* @apiErrorExample UnAuthorizedError Response:
+	*     HTTP/1.1 401 Unauthorized
+	*     {
+	*          "message": "Invalid/missing token"
+	*     }
+	*/
+	router.get('/', chatController.getUserChats);
 
-	// API endpoint to get messages in given chat from messages table
-	// router.get('/:cid', chatController.getMessages);
+	/**
+	* @api {post} /chat Create chat
+	* @apiName CreateChat
+	* @apiGroup Chat
+	*
+	* @apiHeader {String} token Authorization Bearer Token.
+	* @apiParam (Request body) {String} [name] Chat's name.
+	*
+	* @apiSuccess {Number} id Chat ID.
+	* @apiSuccess {String} name Chat name.
+	*
+	* @apiSuccessExample Success Response:
+	*     HTTP/1.1 201 Created
+	*     {
+	*          "id": 1,
+	*          "name": "Chat name"
+	*     }
+	*
+	* @apiError UnauthorizedError Invalid/missing token in authorization header.
+	*
+	* @apiErrorExample UnAuthorizedError Response:
+	*     HTTP/1.1 401 Unauthorized
+	*     {
+	*          "message": "Invalid/missing token"
+	*     }
+	*/
+	router.post('/', chatController.createChat);
+	
+	// must have some middleware to verify user making the API call is part of the given chat
+	router.use('/:cid*', membershipMiddleware);
 
-	// API endpoint to remove given user from given chat from users_chats table
-	// router.delete('/:cid/:uid', chatController.removeUser);
+	/**
+	* @api {get} /chat/:cid Get messages in chat
+	* @apiName GetMessagesInChat
+	* @apiGroup Chat
+	*
+	* @apiParam {Number} cid Chat ID.
+	* @apiHeader {String} token Authorization Bearer Token.
+	*
+	* @apiSuccess {Object[]} - List of chats.
+	* @apiSuccess {String} -.id Chats id.
```

Comment:
Think this is supposed to be `{Number} id Chat ID.` right?
```suggestion
	* @apiSuccess {Number} id Chats ID.
```